cirq-core/cirq/sim/simulator.py

```diff
@@ -822,6 +825,8 @@ def sample_measurement_ops(
                 `MeasurementGate` instances to be sampled form.
             repetitions: The number of samples to take.
             seed: A seed for the pseudorandom number generator.
+            _allow_repeated: If True, adds extra dimension to the result,
+                corresponding to the number of times a key is repeated.
```

Comment:
Right, so this is `StepResult.sample_meaurement_ops`. Since it is public, I didn't want to change the return type or behavior. It's called from `SimulatorBase._run`. Options I considered were: 

* Create a separate method for 3D arrays that allows duplicates
* Skipping this option in `SimulatorBase._run` and having everything go through the full-simulation path (would be much slower this way I believe)
* Just inlining the logic here into `SimulatorBase._run`

This seemed like the most reasonable solution, but either of the others would work too. It would be nice to eventually (or immediately) make the breaking change such that everything returns a 3D array instead of 2D. If we want to continue to offer both 2D and 3D support indefinitely then maybe they should be separate functions instead of parameter-dependent. I'm also fully open to making the parameter non-private.

Reply:
SGTM.